### ISSUE TITLE
TypeError FormBuilder vs SubmitButtonBuilder

### ISSUE THREAD

**Issue by tburschka** [symfony/symfony#26918]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 4.0.8

Hello,

i'm facing currently an TypeError during an unit test, while im try to get a submit button direct via `$this->builder->get('submit')`:

```
TypeError: Return value of Symfony\Component\Form\FormBuilder::resolveChild() must be an instance of Symfony\Component\Form\FormBuilder, instance of Symfony\Component\Form\SubmitButtonBuilder returned
```

There are different instances of FormBuilder (and a submit button should be builded by the SubmitButtonBuilder, but if i call `$this->builder->all()` before `$this->builder->get('submit')`, all unresolvedChildren are created so that the method `resolveChildren`is used (instead the method `resolveChild`) which does not cause the error after the button was created.

**Comment by xabbuh** [comment]
Can you create a small example project that allows to reproduce your issue?

**Comment by xabbuh** [comment]
I am closing here for now, but we can reopen once we have a reproducible example.

**Comment by tburschka** [comment]
@bluser86 thanks for the sample. I don't have much time for it lately and hadn't worked on the project with this error.

**Comment by xabbuh** [comment]
@bluser86 Thanks for the reproducer, will be fixed in #28789.
